### ISSUE TITLE
Paths (clipping) are not resized with the image upon trim

### ISSUE THREAD

**Issue by andreas1974** [dlemstra/Magick.NET#43]
After trimming an image, clipping paths no longer mark the shapes of objects.
For a more detailed explanation, see my question at StackOverflow: http://stackoverflow.com/questions/43280835/how-can-i-trim-an-image-using-magick-net-keeping-clipping-paths-if-they-exist

**Comment by dlemstra** [comment]
Your image contains clipping paths in a vector format. When your image is resized the coordinates in the clipping path are not updated. It will also not be easy to update the svg file yourself. If you want to use the path right after you have resized the image you will need to convert the clipping path to a raster format and resize it yourself. I can create an example for that in a couple days if that is what you want to do. Otherwise I would prefer it if we closed this issue. I can also give an answer to your stackoverflow post in a couple days.

**Comment by andreas1974** [comment]
I just want the path to be kept with the image after that it is trimmed and saved, so that users later on can open the resulting trimmed files and still see and work with their paths in Photoshop and other applications.
I notice that **Resizing** works fine. The path is resized with the image and kept in its normal vector format that Photoshop use.
I'm afraid I don't understand the part in your answer about converting the path to a raster format. I don't intend to work with the path in Magick.Net / ImageMagick... that might be what you are thinking of. I would just like the path to resize with the image, when trimming, just as it already does when resizing.

**Comment by dlemstra** [comment]
I suspect that the path is relative to the size of the image and that is why it works when you resize the image. When you trim the image part of the image is removed and I think you will need to set an x and y offset in the clipping path. I am unable to run some experiments with that right now but I will have some time for this next week. But you might be able to do it yourself. You will need to get the clipping with `string GetAttribute("8BIM:1999,2998:#1")` and set it with `SetAttribute("8BIM:1999,2998:#1", value); and set the x and y offset. I don't know if that is possible but please let me know if you figure out how to do that. I will also introduce a new method in the next release so you don't have to call the GetAttribute and SetAttribute method and can just call `GetClippingPath()` or `GetClippingPath(string name)` (#1 is the name).

**Comment by dlemstra** [comment]
I did some experiments and I could not figure out how to "fix" the clipping path myself. I tried using this: https://www.w3.org/TR/SVG/coords.html#EstablishingANewUserSpace but I could not get it to work. What I haven't tried and will leave as an exercise for the reader is changing the coordinates. I wonder what happens when you reposition the coordinates:

```
M 518.4 151.977 
C 516.48 151.977 514.56 151.977 512.64 151.977
```

And when your image has lost the top 100 pixels subtract the height:

```
M x (y-100)
C x (y-100) x (y-100) x (y-100)
```

```
M 518.4 51.977 
C 516.48 51.977 514.56 51.977 512.64 51.977
```
